goflags.go

```diff
@@ -23,31 +23,43 @@ type FlagSet struct {
 	Marshal     bool
 	description string
 	flagKeys    InsertionOrderedMap
+	groups      []groupData
 }
 
-type flagData struct {
+type groupData struct {
+	name        string
+	description string
+}
+
+type FlagData struct {
```

Comment:
Would it make sense to try to find a way that does not require this struct to be exported, as it rather holds the internal "state" of the flags.

Reply:
I did take a look around, and the best way to make the syntax possible like `StringVar().Group("x")` was to export FlagData and add a method called Group. New ideas around this can be thought of for sure.